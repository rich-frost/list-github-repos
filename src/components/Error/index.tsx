import { Text } from '@chakra-ui/react';

type ErrorProps = {
  error?: string;
};

export default function Error({
  error = 'Sorry for the inconvenience',
}: ErrorProps) {
  return <Text color="tomato">An error occured: {error}</Text>;
}
