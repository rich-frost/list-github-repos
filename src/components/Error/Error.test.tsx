import renderer from 'react-test-renderer';
import Error from './index';

describe('Error component', () => {
  it('renders the component', () => {
    const component = renderer.create(<Error />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the component with custom error', () => {
    const component = renderer.create(
      <Error error="The API failed to respond" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
