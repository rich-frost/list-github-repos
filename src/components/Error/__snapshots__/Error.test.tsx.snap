// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders the component 1`] = `
<p
  className="chakra-text css-tq928n"
>
  An error occured: 
  Sorry for the inconvenience
</p>
`;

exports[`Error component renders the component with custom error 1`] = `
<p
  className="chakra-text css-tq928n"
>
  An error occured: 
  The API failed to respond
</p>
`;
